modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/CurrencySnapshotContextFunctions.scala

```diff
@@ -11,25 +11,35 @@ import scala.util.control.NoStackTrace
 
 import org.tessellation.currency.schema.currency.{CurrencyIncrementalSnapshot, CurrencySnapshotInfo}
 import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.address.Address
 import org.tessellation.sdk.domain.block.processing._
 import org.tessellation.sdk.domain.snapshot.SnapshotContextFunctions
 import org.tessellation.sdk.infrastructure.snapshot.GlobalSnapshotContextFunctions.CannotApplyRewardsError
+import org.tessellation.sdk.infrastructure.snapshot.context._
 import org.tessellation.security.signature.Signed
 
 import derevo.cats.{eqv, show}
 import derevo.derive
 import eu.timepit.refined.auto._
 
-abstract class CurrencySnapshotContextFunctions[F[_]] extends SnapshotContextFunctions[F, CurrencyIncrementalSnapshot, CurrencySnapshotInfo]
+object context {
+  implicit class CurrencySnapshotContextOps(value: (Address, CurrencySnapshotInfo)) {
+    def address: Address = value._1
+    def snapshotInfo: CurrencySnapshotInfo = value._2
+  }
```

Comment:
You used `(Address, CurrencySnapshotInfo)` instead of `case class Context(address: Address, snapshotInfo: CurrencySnapshotInfo)`  but then you added getters to that tuple via implicit class to make it a case class. Imo you could have just used a case class 🤔

Reply:
I cannot use a `case class`. The compilator throws an error about the type tag then.
Also, I cannot use `CurrencySnapshotContext` in `sdk` code, because it's defined in currency L0 module. So in here I have to keep using tuple and the implicits are just for convenience

Reply:
I don't see a reason why we would define a helper `implicit class` and not a helper `case class`. For me it makes no sense to obfuscate it like that. The only matter is where to put it, but if you have a helper implicit class here you could as well have a case class. Also I had changed it locally to a case class and I didn't get any `typeTag` errors.